### PR TITLE
Use proper url in emails sent to pending participants

### LINKED ISSUE #1926: Pending ... notifications: Use ExternalRegistrationURL if local accounts are disabled
Currently we always use `url_for_login` in these emails. Instead, we should use `ExternalRegistrationURL` from the config if `LocalIdentities` are disabled. Maybe add `url_for_register` util that points to either the registration page or the external url (or, if neither are set, to the login page as a fallback)?

### PR DESCRIPTION
**Pull request by mic4ael**
fixes #1926

### CODE REVIEW

File: indico/MaKaC/common/pendingQueues.py [hunk 1]
```diff
@@ -97,6 +97,11 @@ def _calculateParticipationsByConf(self):
                 d[conf] = [ps]
         return d
 
+    def _get_url(self):
+        local_identities = Config.getInstance().getLocalIdentities()
```
Comment (ThiefMaster) [inline indico/MaKaC/common/pendingQueues.py]: why not do that in `url_for_register()`?

File: indico/modules/auth/util.py [hunk 2]
```diff
@@ -69,3 +70,12 @@ def url_for_login(next_url=None):
...
 def url_for_logout(next_url=None):
     return url_for('auth.logout', next=next_url)
+
+
+def url_for_register(next_url=None):
+    local_identities_enabled = Config.getInstance().getLocalIdentities()
+    if local_identities_enabled:
+        return url_for('auth.register', next=next_url)
+
+    external_url = Config.getInstance().getExternalRegistrationurl()
```
Comment (ThiefMaster) [inline indico/modules/auth/util.py]: typo - should be `getExternalRegistrationURL()`

Comment (mic4ael) [inline indico/modules/auth/util.py]: Thanks! I didn't notice it before!

File: indico/modules/auth/util.py [hunk 3]
```diff
@@ -20,6 +20,7 @@
 from werkzeug.datastructures import MultiDict
 
 from indico.web.flask.util import url_for
+from indico.core.config import Config
```
Comment (ThiefMaster) [inline indico/modules/auth/util.py]: `'c' < 'w'`

File: indico/modules/auth/util.py [hunk 4]
```diff
@@ -69,3 +70,12 @@ def url_for_login(next_url=None):
 
 def url_for_logout(next_url=None):
     return url_for('auth.logout', next=next_url)
+
+
+def url_for_register(next_url=None):
+    local_identities_enabled = Config.getInstance().getLocalIdentities()
```
Comment (ThiefMaster) [inline indico/modules/auth/util.py]: used only once, so you could use `if Config.getInstance().getLocalIdentities():` directly

File: indico/modules/auth/util.py [hunk 5]
```diff
@@ -69,3 +70,12 @@ def url_for_login(next_url=None):
...
+    if local_identities_enabled:
+        return url_for('auth.register', next=next_url)
+
+    external_url = Config.getInstance().getExternalRegistrationUrl()
```
Comment (ThiefMaster) [inline indico/modules/auth/util.py]: this is still wrong, it's `getExternalRegistrationURL()`...

Comment (mic4ael) [inline indico/modules/auth/util.py]: Drawbacks of using Sublime for development..

Comment (ThiefMaster) [inline indico/modules/auth/util.py]: An IDEs wouldn't really help here; this is how those ugly getters are implemented:

    def __getattr__(self, attr):
        if attr[0:3] == 'get':
            def configFinder(k):
                return self._configVars[k]
            return lambda: configFinder(attr[3:])
        else:
            raise AttributeError

Comment (mic4ael) [inline indico/modules/auth/util.py]: Yep, that is why I deleted the comment right away ;)

**Comment by pferreir** [comment]
Merged in dedf66ffdea9f1c58a6a82fdad9b8863be4202f4.

**Pull request closed without merging**